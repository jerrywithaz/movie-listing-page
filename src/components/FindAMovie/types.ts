import { Cinema, SelectedCinema } from "core/types";

export type FindAMovieProps = {
    theaters: Cinema[],
    selectedTheater: SelectedCinema,
    setSelectedTheater: (cinema: Cinema) => void
} & React.HTMLAttributes<HTMLDivElement>;