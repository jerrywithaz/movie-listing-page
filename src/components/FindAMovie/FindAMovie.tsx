import React from 'react';
import Box from 'components/Box';
import { FindAMovieProps } from './types';

const FindAMovie = (props: FindAMovieProps) => {

    const width = `${(752/1440) * 100}%`;
    const height = `${(722/1440) * 100}%`;

    return (
        <Box {...props} width={width} height={height}>

        </Box>
    );
};

export default FindAMovie;