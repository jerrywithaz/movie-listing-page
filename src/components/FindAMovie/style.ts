import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    width: 560px;
    margin: 0 auto;
    max-width: 100%;
    height: 100%;
    overflow: hidden;
`;