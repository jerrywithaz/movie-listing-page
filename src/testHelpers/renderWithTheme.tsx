import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;