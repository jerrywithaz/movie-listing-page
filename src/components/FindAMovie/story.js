import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import FindAMovie from './';
import { getMockFilms, getMockTheaters } from 'core/mocks';

const story = storiesOf('FindAMovie', module);

const films = getMockFilms();
const theaters = getMockTheaters();
const selectedTheater = theaters[0];

story.add('FindAMovie', () => (
    <FindAMovie
        films={films}
        setSelectedTheater={action("setSelectedTheater")} 
        theaters={theaters} 
        selectedTheater={selectedTheater}/>
));
  