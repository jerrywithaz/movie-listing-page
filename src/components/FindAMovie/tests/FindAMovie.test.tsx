import React from 'react';
import FindAMovie from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';

describe("components/FindAMovie", () => {

    it("should match snapshot", () => {

        const tree = renderWithTheme(
            <FindAMovie 
                theaters={[]} 
                selectedTheater={null}
                setSelectedTheater={() => {}}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

});