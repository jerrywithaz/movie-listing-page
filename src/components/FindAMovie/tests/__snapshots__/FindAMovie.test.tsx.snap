// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FindAMovie should match snapshot 1`] = `
.c3 {
  font-size: 26px;
  line-height: 30px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  color: #000000;
  text-transform: uppercase;
  margin: 0px 0px 24px 0px;
}

.c6 {
  font-size: 16px;
  line-height: 20px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #6B6B6B;
  text-transform: uppercase;
  margin: 24px 0px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 196px;
  margin-bottom: 32px;
  overflow: hidden;
}

.c7 {
  margin-bottom: 16px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
}

.c0 {
  height: 722px;
  width: 752px;
  box-shadow: 0px 1px 5px rgba(24,17,3,0.2), 0px 3px 4px rgba(0,0,0,0.12), 0px 2px 4px rgba(24,17,3,0.14);
  border-radius: 4px;
  padding: 32px;
  background: #ffffff;
}

.c4 {
  display: block;
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 560px;
  margin: 0 auto;
  max-width: 100%;
  height: 100%;
  overflow: hidden;
}

@media (max-width:992px) {
  .c1 {
    max-width: calc(100% - 64px);
  }
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <h1
      className="c3 c4"
    >
      Find a Movie
    </h1>
    <div
      className="c5"
    >
      <h2
        className="c6 c7"
      >
        Select Theater
      </h2>
      <div
        className="c8"
      />
    </div>
  </div>
</div>
`;
